components/mpas-ocean/src/shared/mpas_ocn_vmix.F

```diff
@@ -463,15 +467,18 @@ subroutine ocn_vel_vmix_tend_implicit(dt, kineticEnergyCell, vertViscTopOfEdge,
          ! one active layer
          if (N .eq. Nsurf) then
             normalVelocity(N,iEdge) = normalVelocity(N,iEdge)  &
-                / (1.0_RKIND + dt*implicitBottomDragCoef &
+                / (1.0_RKIND + dt*(implicitTopDragCoef * landIceEdgeFraction(iEdge) + implicitBottomDragCoef) &
                    * sqrt(kineticEnergyCell(N,cell1) + kineticEnergyCell(N,cell2)) / layerThickEdgeMean(N,iEdge) )
          else
 
            ! tridiagonal matrix algorithm
            C(Nsurf)     = -2.0_RKIND*dt*vertViscTopOfEdge(Nsurf+1,iEdge) &
                       / (layerThickEdgeMean(Nsurf,iEdge) + layerThickEdgeMean(Nsurf+1,iEdge)) &
                       / layerThickEdgeMean(Nsurf,iEdge)
-           bTemp(Nsurf) = 1.0_RKIND - C(Nsurf)
+           bTemp(Nsurf) = 1.0_RKIND - C(Nsurf) &
+              + dt * implicitTopDragCoef * landIceEdgeFraction(iEdge) &
+                * sqrt(kineticEnergyCell(Nsurf,cell1) + kineticEnergyCell(Nsurf,cell2)) / &
+                layerThickEdgeMean(Nsurf,iEdge)
```

Comment:
This is where the action happens. I can see from my [original design document a long time ago](https://github.com/MPAS-Dev/MPAS-Documents/blob/master/ocean/implicit_vert_mix/implicit_vert_diff_design.pdf), that you can add top drag in the same way as bottom drag:
![image](https://user-images.githubusercontent.com/4036816/221951656-bfca11cc-752e-4a52-92ce-53230861a509.png)
So in (3.42) above, you would add another equation for k=N_{surf}, where 
```
B_k = 1 - C_k  + \delta t * c_{drag} |u^n| / h^{n+1}
```
which is the line above. So that looks correct.

I do have a concern that `implicitBottomDragCoef` appears in line [499](https://github.com/E3SM-Project/E3SM/pull/5447/files#diff-c04cc906ba019d7e415af5dfef477ce54bb2e70ad8d59a23a35a58fe2b771787L499) (now 506) below in the backsolve, and there is no corresponding `implicitTopDragCoef`. I think that is because it is already in bTemp in line 504 (now 511), here
```
   normalVelocity(k,iEdge) = (rTemp(k) - C(k)*normalVelocity(k+1,iEdge)) / bTemp(k)
```
in the final solve for `normalVelocity`. 

@cbegeman can you just communicate your design and thought process here? I just want to make sure you worked through the algebra in the design document again.

Reply:
@mark-petersen Thanks for your question and the thought you're giving this. I'm realizing now that I could have saved myself a bit of time by asking you for the design doc instead of tracking the terms through the code!

So as you've seen, in the single layer case, this drag term is just substituted directly into the `normalVelocity` solve on current line 470 rather than being stored in `bTemp`.

For the multi-layer case, I opted to include the implicit top drag only in `bTemp(Nsurf)` because this is the top layer. This particular drag formulation is only applied to the top layer, so the drag term does not appear in `bTemp(k)` where `k` ranges from `Nsurf+1` to `N-1`. The line that you pointed out, currently line 506, is where the bottom drag term is substituted directly into the `normalVelocity` solve for the bottom layer. As a reminder `N = maxLevelEdgeTop`. Since top drag is only applied to the top layer, we also don't need to include `implicitTopDragCoef` here at the `N` level.

Does that answer your question?

Reply:
Yes, thanks. By the time I had finished writing my (lengthy) comment, I also came to that conclusion. But I thought I would post it anyway!